### PR TITLE
Message on Save State (fix for issue #1783)

### PR DESCRIPTION
**Pull request by b10z**
This fix displays a popup message (as mentioned here #1783) the first time the Save State button is pressed. 
Tested on Android (!only!, i couldn't get visual studio build to work as expected for some reason) and works fine for me. if there is any issue please let me know :D

### CODE REVIEW

File: libswirl/types.h [hunk 1]
```diff
@@ -360,8 +360,7 @@ using namespace std;
 #endif
 
 
-// NOTE: Always inline for macOS builds or it causes duplicate symbol linker errors
-#if DEBUG && HOST_OS != OS_DARWIN
+#if DEBUG
```
Comment (skmp) [inline libswirl/types.h]: Can you revert this?  it might break the osx build

**Comment by skmp** [comment]
Looks good overall, can you please revert the #ifdef so that other platforms aren't affected?

File: libswirl/types.h (hunk 1 above)
Comment (b10z) [reply libswirl/types.h]: sure thing! i didn't even notice that change.. its fixed now i suppose! :D

**Review by skmp — APPROVED** [review]

**Comment by skmp** [comment]
Thx for the contribution! 🏍️

**Pull request merged**